gcc/common/config/riscv/riscv-common.c

```diff
@@ -51,25 +51,27 @@ struct riscv_implied_info_t
 {
   const char *ext;
   const char *implied_ext;
+  int xlen;
 };
 
 /* Implied ISA info, must end with NULL sentinel.  */
 riscv_implied_info_t riscv_implied_info[] =
 {
-  {"d", "f"},
-  {"k", "zkn"},
-  {"k", "zkr"},
-  {"zkn", "zkne"},
-  {"zkn", "zknd"},
-  {"zkn", "zknh"},
-  {"zkn", "zkg"},
-  {"zkn", "zkb"},
-  {"zks", "zksed"},
-  {"zks", "zksh"},
-  {"zks", "zkg"},
-  {"zks", "zkb"},
-  {"p", "zpn"},
-  {"p", "zpsf"},
+  {"d", "f", 0},
+  {"k", "zkn", 0},
+  {"k", "zkr", 0},
+  {"zkn", "zkne", 0},
+  {"zkn", "zknd", 0},
+  {"zkn", "zknh", 0},
+  {"zkn", "zkg", 0},
+  {"zkn", "zkb", 0},
+  {"zks", "zksed", 0},
+  {"zks", "zksh", 0},
+  {"zks", "zkg", 0},
+  {"zks", "zkb", 0},
+  {"p", "zpn", 0},
+  {"p", "zprv", 64},
```

Comment:
I am not sure if xlen should be added here.
Look for example at Zknh from k-ext.
It has 4 sha256* insns for both rv32 and rv64, 6 sha512* rv32-only insns and 4 sha512* rv64-only insns.
If we do follow this logic, Zprv just have an empty quantity of rv32-suitable insns (i.e. all its insns are rv64+-only).
This actually rises a question if Zprv is needed at all -- I asked it [here](https://github.com/riscv/riscv-p-spec/issues/76).